modules/deployment/code_build.tf

```diff
@@ -0,0 +1,193 @@
+resource "aws_cloudwatch_log_group" "this" {
+  name              = "/aws/codebuild/${var.function_name}-deployment"
+  retention_in_days = var.cloudwatch_logs_retention_in_days
+  tags              = var.tags
+}
+
+resource "aws_codebuild_project" "this" {
+  name         = "${var.function_name}-deployment"
+  service_role = var.code_build_role_arn == "" ? aws_iam_role.code_build_role[0].arn : var.code_build_role_arn
+  tags         = var.tags
+
+  artifacts {
+    type                = "CODEPIPELINE"
+    artifact_identifier = "deploy_output"
+    location            = "appspec.json"
```

Comment:
this file is technically not actually written nor archived anywhere. I've tried to create this file and pass it to code deploy, ... but no.